types/credentials.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) OpenFaaS Author(s). All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package types
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+)
+
+// BasicAuthCredentials are the user and password strings
+type BasicAuthCredentials struct {
+	User     string
+	Password string
+}
+
+// ReadBasicAuth interface for reading the credentials
+type ReadBasicAuth interface {
+	Read() (error, *BasicAuthCredentials)
+}
+
+// ReadBasicAuthFromDisk defines the path for the secret
+type ReadBasicAuthFromDisk struct {
+	SecretMountPath string
+}
+
+func (r *ReadBasicAuthFromDisk) Read() (*BasicAuthCredentials, error) {
+	var credentials *BasicAuthCredentials
+
+	if len(r.SecretMountPath) == 0 {
+		return nil, fmt.Errorf("invalid SecretMountPath specified for reading secrets")
+	}
+
+	userPath := path.Join(r.SecretMountPath, "basic-auth-user")
+	user, userErr := ioutil.ReadFile(userPath)
+	if userErr != nil {
+		return nil, fmt.Errorf("unable to load %s", userPath)
+	}
+
+	userPassword := path.Join(r.SecretMountPath, "basic-auth-password")
+	password, passErr := ioutil.ReadFile(userPassword)
+	if passErr != nil {
+		return nil, fmt.Errorf("Unable to load %s", userPassword)
+	}
+
+	credentials = &BasicAuthCredentials{
```

Comment:
Should we handle potential new lines here? Genuine question as I'm 50/50 on it but we have had scenarios where secret creation has gone awry due to various methods implicitly adding \r